apps/chat/src/store/publication/publication.epics.ts

```diff
@@ -1348,30 +1357,182 @@ const uploadAllPublishedWithMeItemsFailEpic: AppEpic = (action$) =>
     ),
   );
 
-const updatePublicationRequestEpic: AppEpic = (action$) =>
+const updatePublicationRequestEpic: AppEpic = (action$, state$) =>
   action$.pipe(
     ofType(PublicationActions.updatePublicationRequest.type),
     switchMap(({ payload }) =>
       forkJoin({
         payload: processPublicationResources(payload.dataToUpdate),
-        url: payload.url,
+        url: of(payload.url),
       }),
     ),
     switchMap(({ payload, url }) => {
-      const { publicationData, isPublishingExternalFiles } = payload;
-      if (isPublishingExternalFiles) {
-        return of(
-          PublicationActions.publishFail(
-            errorsMessages.publicationWithExternalFilesFailed,
-          ),
-        );
-      }
+      const { publicationData } = payload;
 
       return PublicationService.updatePublicationRequest({
         publicationData,
         url,
       }).pipe(
-        switchMap(() => EMPTY),
+        switchMap((response) => {
+          const oldPublicationResources =
+            PublicationSelectors.selectPublicationByUrl(state$.value, url)
+              ?.resources ?? [];
+          const newPublicationResources = response.resources.map(
+            (resource) => ({
+              ...resource,
+              sourceUrl: ApiUtils.decodeApiUrl(resource.sourceUrl ?? ''),
+              targetUrl: ApiUtils.decodeApiUrl(resource.targetUrl),
+              reviewUrl: ApiUtils.decodeApiUrl(resource.reviewUrl),
+            }),
+          );
+
+          const resourcesRequiresUpdate = newPublicationResources.filter(
+            (newResource) => {
+              const match = oldPublicationResources.find(
+                (oldResource) =>
+                  oldResource?.sourceUrl === newResource.sourceUrl,
+              );
+
+              return match && match.targetUrl !== newResource.targetUrl;
+            },
+          );
+
+          const resourcesRequiresUpdateIds = resourcesRequiresUpdate.map(
+            (resource) => resource.reviewUrl,
+          );
+          const conversationsRequiresUpdate = resourcesRequiresUpdateIds.filter(
+            (id) => isConversationId(id),
+          );
+          const promptsRequiresUpdate = resourcesRequiresUpdateIds.filter(
+            (id) => isPromptId(id),
+          );
+          const applicationsRequiresUpdate = resourcesRequiresUpdateIds.filter(
+            (id) => isApplicationId(id),
+          );
```

Comment:
What do you think if we change like this:
```suggestion
        const orderedResourcesRequiresUpdateIds: {
        conversationsRequiresUpdate: string[];
        promptsRequiresUpdate: string[];
        applicationsRequiresUpdate: string[];
      } = {
        conversationsRequiresUpdate: [],
        promptsRequiresUpdate: [],
        applicationsRequiresUpdate: [],
      };
      resourcesRequiresUpdateIds.forEach((id) => {
        if (isConversationId(id)) {
          orderedResourcesRequiresUpdateIds.conversationsRequiresUpdate.push(
            id,
          );
        } else if (isPromptId(id)) {
          orderedResourcesRequiresUpdateIds.promptsRequiresUpdate.push(id);
        } else if (isApplicationId(id)) {
          orderedResourcesRequiresUpdateIds.applicationsRequiresUpdate.push(id);
        }
      });
          
```